logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreFailure.kt

```diff
@@ -96,3 +99,12 @@ internal inline fun <T : Any> wrapStorageRequest(storageRequest: () -> T?): Eith
         Either.Left(StorageFailure.Generic(e))
     }
 }
+
+internal inline fun <T : Any> wrapFlowStorageRequest(storageRequest: () -> Flow<T?>): Flow<Either<StorageFailure, T>> {
+    return try {
+        storageRequest().map { it?.let { data -> Either.Right(data) } ?: Either.Left(StorageFailure.DataNotFound) }
+    } catch (e: Exception) {
+        kaliumLogger.e(e.stackTraceToString())
+        flowOf(Either.Left(StorageFailure.Generic(e)))
+    }
```

Comment:
There is a `catch` operator so I'm wondering if something like that would work:
```
return storageRequest()
    .map { it?.let { data -> Either.Right(data) } ?: Either.Left(StorageFailure.DataNotFound) }
    .catch { e -> 
        kaliumLogger.e(e.stackTraceToString())
        emit(Either.Left(StorageFailure.Generic(e)))
     }
```
this way we don't create a separate Flow in case of exception 😄